src/main/java/ru/yandex/practicum/filmorate/dao/FilmDbStorageDao.java

```diff
@@ -0,0 +1,122 @@
+package ru.yandex.practicum.filmorate.dao;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+import ru.yandex.practicum.filmorate.exceptions.DataNotFoundException;
+import ru.yandex.practicum.filmorate.model.Film;
+import ru.yandex.practicum.filmorate.model.Mpa;
+import ru.yandex.practicum.filmorate.service.FilmGenreService;
+import ru.yandex.practicum.filmorate.service.MpaService;
+import ru.yandex.practicum.filmorate.storage.film.FilmStorage;
+
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.*;
+
+@Component
+@Slf4j
+@Qualifier("FilmDbStorage")
+@RequiredArgsConstructor
+public class FilmDbStorageDao implements FilmStorage {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final FilmGenreService filmGenreService;
+    private final MpaService mpaService;
+
+    @Override
+    public Film findFilm(int id) {
+        String sql = "SELECT * FROM FILMS WHERE FILM_ID = ?";
```

Comment:
final String sqlQuery = "select * from FILMS f, MPA m where f.MPA_ID = m.MPA_ID AND FILM_ID = ?";